diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/SingleLineStatementsRuleWarnTest.kt

```diff
@@ -0,0 +1,140 @@
+        )
+    }
+
+    @Test
+    fun `check two in one line without space`() {
+        lintMethod(SingleLineStatementsRule(),
+                """
+                    |fun foo() {
+                    |   if (x > 0){
+                    |       goo()
+                    |   }; else { print(123) }
+                    |}
+                """.trimMargin(),
+                LintError(4,5,ruleId,"${MORE_THAN_ONE_STATEMENT_PER_LINE.warnText()}    }; else { print(123) }", false)
+        )
+    }
+
+    @Test
+    fun `check correct test without more than one statement`() {
+        lintMethod(SingleLineStatementsRule(),
+                """
+                    |import com.pinterest.ktlint.core.KtLint
+                    |
+                    |fun foo() {
+                    |    if (x < -5) {
+                    |       goo();
+                    |    }
+                    |    else {
+                    |    }
+                    |    
+                    |    when(x) {
+                    |       1 -> println(1)
+                    |       else -> println("3;5")
+                    |    }
+                    |    val a = 5
+                    |    println(1)
+                    |}
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    fun `check semicolon with enum class expression`() {
+        lintMethod(SingleLineStatementsRule(),
+                """
+                    |enum class ProtocolState {
+                    |   WAITING {
+                    |       override fun signal() = TALKING
+                    |   },
+                    |   TALKING {
+                    |       override fun signal() = WAITING
+                    |   }; abstract fun signal(): ProtocolState
+                    |}
+                """.trimMargin(),
+                LintError(7,5,ruleId,"${MORE_THAN_ONE_STATEMENT_PER_LINE.warnText()}    }; abstract fun signal(): ProtocolState", false)
+        )
+    }
+
+    @Test
+    fun `check it expression with two wrong seminolon`() {
+        lintMethod(SingleLineStatementsRule(),
+                """
+                    |fun foo() {
+                    |   if(x > 0) {
+                    |       if ( y> 0){
+                    |           ji()
+                    |       }; gt()
+                    |   }; gr()
+                    |}
+                """.trimMargin(),
+                LintError(5,9,ruleId,"${MORE_THAN_ONE_STATEMENT_PER_LINE.warnText()}        }; gt()", false),
+                LintError(6,5,ruleId,"${MORE_THAN_ONE_STATEMENT_PER_LINE.warnText()}    }; gr()", false)
+        )
+    }
+
+    @Test
+    fun `check semicolon in the beginning of the line`() {
+        lintMethod(SingleLineStatementsRule(),
+                """
+                    |fun foo() {
+                    |   ; grr()
+                    |}
+                """.trimMargin(),
+                LintError(2,4,ruleId,"${MORE_THAN_ONE_STATEMENT_PER_LINE.warnText()}    ; grr()", false)
```

Comment:
Please add this example to FixTest as well. Will you still insert newline even though there is only one statement?